src/auth/ducks/reducers.ts

```diff
@@ -0,0 +1,42 @@
+import { UserAuthenticationReducerState } from './types';
+import { authenticateUser, UserAuthResponse } from './actions';
+import { C4CAction } from '../../store';
+import {
+  ASYNC_REQUEST_FAILED_ACTION,
+  ASYNC_REQUEST_LOADED_ACTION,
+  ASYNC_REQUEST_LOADING_ACTION,
+  AsyncRequestNotStarted,
+  generateAsyncRequestReducer,
+} from '../../utils/asyncRequest';
+
+export const initialUserState: UserAuthenticationReducerState = {
+  userAuthenticationDetails: AsyncRequestNotStarted<UserAuthResponse, void>(),
+};
+
+const userAuthenticationRequestReducer = generateAsyncRequestReducer<
+  UserAuthenticationReducerState,
+  UserAuthResponse,
+  void
+>(authenticateUser.key);
+
+const reducers = (
```

Comment:
Is this the top level reducer? If so, would all API calls that get data have to be included in this down the line?

Reply:
Everything in this `auth/ducks/` folder will be solely for authentication-related data and actions. Each container typically will have its own ducks. The reducers, along with the action types, are combined in `store.ts`.